doc/introduction/circuits.rst

```diff
@@ -171,6 +175,50 @@ for accessing the results of your program execution:
     will need to increase the cutoff dimension.
 
 
+Symbolic parameters
+-------------------
+
+The quantum operations can take both numerical and symbolic parameters. The latter fall into two types:
+
+* **Measured parameters**: Certain quantum programs require that
+  operations can be conditioned on measurement results obtained during the execution of the
+  program. In this case the parameter value is not known until the measurement is made
+  (or simulated). The latest measurement result of qumode ``i`` is referred to as ``q[i].par``.
+
+* **Free parameters**: A *parametrized circuit template* is a program that
+  depends on a number of free parameters. These parameters are bound to new fixed
+  values each time the program is executed.
+  The free parameters are created accessed using the :meth:`Program.params` method.
+
+The symbolic parameters can be combined and transformed using basic algebraic operations and
+common mathematical functions in the ``strawberryfields.parameters.parfuncs`` namespace.
```

Comment:
I'm not sure I really like the name `parfuncs`. At the same time, the `strawberryfields.parameters` module is not  user-facing apart from this one import.

I think we should import this to top level, and give it a better name, how about the following?

```python
ops.Sgate(1 - sf.math.sin(q[0].par))
```

This matches nicely with TF 2.0, where a lot of the previous top level functions are now in the `math` module (i.e., `tf.math.log`).

Reply:
Would `sf.math` be a module, or would `math` be a `SimpleNamespace` object inside `sf`? The latter is how `parfuncs` works now.

Reply:
I was imagining just a `SimpleNamespace` object. However, it would be nice if documentation and autocomplete could continue to work, maybe it needs `__dir__` defined?

Reply:
Looks like a `SimpleNamespace` object works nicely, it supports autocompletion and the docs are inherited from the wrapped Sympy funcs.

Reply:
Okay, it's done.